lisa/env.py

```diff
@@ -210,6 +212,83 @@ def from_one_conf(cls, path):
             plat_info = None
         return cls(target_conf=target_conf, plat_info=plat_info)
 
+    @classmethod
+    def from_cli(cls, argv=None):
+        """
+        Create a TestEnv from command line arguments.
+
+        :param argv: The list of arguments. ``sys.argv[1:]`` will be used if
+          this is ``None``.
+        :type argv: list(str)
+
+        Trying to use this in a script that expects extra arguments is bound
+        to be confusing (help message woes, argument clashes...), so for now
+        this should only be used in scripts that only expect TestEnv args.
+
+        **Using the ``argv`` parameter**::
+
+            import shlex
+            from lisa.env import TestEnv
+
+            cmd = "--kind linux -n 192.168.0.1 -u foo -p bar"
+            te = TestEnv.from_cli(shlex.split(cmd))
+        """
+        # Subparsers cannot let us specify --kind=android, at best we could
+        # have --android which is lousy. Instead, use a first parser to figure
+        # out the target kind, then create a new parser for that specific kind.
+        kind_parser = argparse.ArgumentParser(
+            # Disable the automatic help to not catch e.g. ./script.py -k linux -h
+            add_help=False,
+            formatter_class=argparse.RawDescriptionHelpFormatter,
+            description=textwrap.dedent(
+                """
+                Extra arguments differ depending on the value passed to 'kind'.
+                Try e.g. \"--kind android -h\" to see the arguments for android targets.
```

Comment:
\" can be turned into " when inside a triple-quoted string or inside single quotes